src/routes/docs/contributing/code-style.svx

```diff
@@ -10,32 +10,97 @@
   
 # Code Style
 
-1. **Use** `PascalCase`:
+To make the codebase consistent and easy to understand, we require you to follow our predefined code style rules:
 
-- class names
-- method names
-- const names
+1. Naming notation for objects and functions
+- Use PascalCase for classes
+- Use PascalCase prefixed with 'I' for interfaces
+- Use PascalCase for structs
+- Use PascalCase for functions/methods
+- Use "Base" prefix for base classes
+- Use "Service" suffix for service classes and interfaces
+- Use "Model" suffix for model classes
+- Use "ViewModel" suffix for view model classes
+- Use "Dialog" suffix for dialog controls
+- Use "Control" suffix for user controls
+- Use "Page" suffix for pages
 
-2. **Use** `camelCase`:
+2. Naming notation for variables
+- Use PascalCase for private/protected/public static fields/properties
+- Use PascalCase for properties
+- Use PascalCase prefixed with '_' for private fields backing properties
+- Use camelCase for protected and public fields
+- Use camelCase for protected fields backing properties
+- Use camelCase prefixed with '_' for private fields
+- Use camelCase for method parameters
+- Use Hungarian Notation only for OS functions/API calls
+- Use UPPER_CASE for constant variables
+- Use predefined primitive types like `int`, `long`, `string` instead of `Int32`, `Int64`, `String`
 
-- method arguments
-- local variables
-- private fields
+3. Code readability and clarity
+- Use curly brackets for single line `if`, `for` and `foreach` statements.'
+- Do not use `#region` directives
+- Align `while();` on the same line of curly bracket of `do..while`
+- Use the prefixes `Is`, `Has`, `Have`, `Any`, `Can` or similar keywords for boolean names and method names if applicable.
+- Do not use `Enum` or `Flag(s)` suffixes/prefixes for Enum names.
+- Align curly brackets on new lines
+- Fields, properties, methods and constructors must be separated by one line
+  except for property backing fields that are not separated by any amount of lines
+- Do not manually invoke event handling functions or commands
+- Do not use `public` modifiers in interface's properties and functions
+- Do not prefix interface name before implementing function/property unless necessary
+- Define constant types in `Constants` class following its structure
+- Mark classes as `abstract` or `sealed` if applicable.
+- Use single line return `=>` for simple functions
+- Complex classes and interfaces must be documented
```

Comment:
Not sure if that guideline/rule would help much since "complex" is quite debatable.